src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,113 @@
+            Renderable.string("§d§lTaming 60 Cost: §r§d(${
+                if (config.otherRarities) "cheapest" else "exact"
+            } rarity)")
+        )
+        var totalCost = 0.0
+        for (line in lore) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val origPetString = group("pet")
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val originalTier = LorenzRarity.entries.find { it.name == origTierString.uppercase() }!!.id
+                val (petPrice, cheapestTier) = findCheapestTier(pet, originalTier)
+                val displayPetString =
+                    if (cheapestTier == originalTier) group("fullThing")
+                    else "${LorenzRarity.entries.find { it.id == cheapestTier }!!.formattedName} $origPetString"
+                if (petPrice != -1.0) {
+                    totalCost += petPrice
+                    updateList.add(Renderable.clickAndHover(
+                        text = " §7- $displayPetString§7: §6${petPrice.addSeparators()} coins",
+                        tips = listOf(
+                            "§aClick to run §e/ahs ] $origPetString §ato find it on the Auction House.",
+                            "§aNotes: §eSet the rarity filter yourself. §cBooster Cookie required!"
+                        ),
+                        onClick = { LorenzUtils.sendCommandToServer("ahs ] $origPetString") }
+                    ))
+                } else {
+                    updateList.add(Renderable.clickAndHover(
+                        text = " §7- $displayPetString§7: §eNot on AH; view its Wiki article here.",
+                        tips = listOf("§4Click to run §e/wiki $pet §4to view how to obtain it."),
+                        onClick = { LorenzUtils.sendCommandToServer("wiki $pet") }
+                    ))
+                }
+            }
+        }
+        updateList.add(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"))
+        if (config.otherRarities) updateList.add(Renderable.string("§c§lDisclaimer:§r§c Total does not include costs to upgrade via Kat."))
+        return updateList
+    }
+
+    private fun findCheapestTier(pet: String, originalTier: Int): Pair<Double, Int> {
+        val petPriceOne = "$pet$SEPARATOR$originalTier".getPetPrice()
+        val petPrices: MutableList<Double> = mutableListOf(petPriceOne)
+        if (config.otherRarities || petPriceOne == -1.0) {
+            petPrices.add("$pet$SEPARATOR${originalTier - 1}".getPetPrice(otherRarity = true))
+            if (originalTier != 5) petPrices.add("$pet$SEPARATOR${originalTier + 1}".getPetPrice(otherRarity = true))
+        }
+        val petPrice = petPrices.min()
+        val cheapestTier = petPrices.cheapestTierIndex(petPrice, originalTier)
+        return Pair(petPrice, cheapestTier)
```

Comment:
replace it with
```suggestion
return petPrices.withIndex().minBy { it.value }
```
and delete the cheapestTierIndex function.

Reply:
broke the rarity display in the HUD, reverted

Reply:
the order might be wrong. It is index, value but I think you have value, index

Reply:
it's not a matter of the L to R order of value declarations, it's that the HUD is now showing me ~53M coins for a *common* enderman [use this artifact and see for yourself](https://github.com/RayDeeUx/SkyHanni/actions/runs/7603597902)

(to see buildList fail at making the HUD show up, use this artifact)[https://github.com/RayDeeUx/SkyHanni/actions/runs/7603478706]

Reply:
idk, why this happends but it is not the cause of the changes. I used the version from https://github.com/hannibal002/SkyHanni/pull/934/commits/eace810048e2da55016525880134dfa90356ecc2 and got 53m as well.
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/b71639e1-ad5f-40e9-93c1-dc0639fbce09)
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/6195aa34-6036-472c-ba23-48b66303694c)

Reply:
compromise:
```kt
val tierAndIndex = petPrices.withIndex().minBy { it.value }
return Pair(tierAndIndex.cheapestTierIndex(originalTier), tierAndIndex.value)
```
```kt
private fun IndexedValue<Double>.cheapestTierIndex(originalTier: Int) = if (this.index == 1) originalTier - 1 else if (this.index == 2) originalTier + 1 else originalTier
```

Reply:
I found the bug. Sorry my fault. Didn't look at the order of stuff. The compromise is good just change the function to a when instead of an if then it is fine.